core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3213,7 +3213,14 @@ public void activeSyncMetadata(AlluxioURI path, Collection<AlluxioURI> changedFi
 
     Map<AlluxioURI, UfsStatus> statusCache;
     try (RpcContext rpcContext = createRpcContext()) {
-      statusCache = populateStatusCache(path, DescendantType.ALL);
+      if (changedFiles == null) {
+        // full sync
```

Comment:
should we just move these comments and statusCache population into the next set of `if` statements?